src/LinuxTracing/SwitchesStatesNamesVisitor.cpp

```diff
@@ -63,7 +64,13 @@ bool SwitchesStatesNamesVisitor::TidMatchesPidFilter(pid_t tid) {
     return false;
   }
 
-  return tid_to_pid_it->second == thread_state_pid_filter_;
+  for (const pid_t thread_state_pid_filter : thread_state_pid_filters_) {
```

Comment:
Consider making `thread_state_pid_filters_` a  `absl::flat_hash_set`. This would become a call to `.contains()`. It also might be faster (or not, depending on the typical number of elements in the filter).

Reply:
Yeah that's right, this is done on purpose for performance.